autotest/autotest.cpp

```diff
@@ -277,20 +277,43 @@ void tAutotest::sendThread(std::string interfaceName,
 	pcap_close(pcap);
 }
 
-static bool readPacket(int fd, pcap_pkthdr* header, u_char* data)
+bool readTimeLimited(int fd, void* data, ssize_t len, std::chrono::system_clock::time_point time_to_give_up)
 {
 	ssize_t ret = 0;
-	int i;
-
-	struct packHeader hdr;
-	for (i = 0; i < 1000; ++i)
+	char* buf = reinterpret_cast<char*>(data);
```

Comment:
I don't like `void*`..
IMO it's better to ensure type safety with templated function and overloading:
```cpp
bool readTimeLimited(int fd, char* data, ssize_t len, std::chrono::system_clock::time_point time_to_give_up)
{
    ssize_t ret = 0;
    while (len > 0 && std::chrono::system_clock::now() < time_to_give_up)
    {
        ret = read(fd, data, len);
        switch (ret)
        {
            case 0:
                return false;
            case -1:
                if (errno == EAGAIN || errno == EWOULDBLOCK)
                {
                    std::this_thread::sleep_for(READ_WAIT_UNTIL_RETRY);
                }
                else
                {
                    return false;
                }
                break;
            default:
                len -= ret;
                data += ret;
                break;
        }
    }
    return len == 0;
}

template <typename T>
bool readTimeLimited(int fd, T& data, std::chrono::system_clock::time_point time_to_give_up)
{
    return readTimeLimited(fd, reinterpret_cast<char*>(&data), sizeof(T), time_to_give_up);
}
```